Generate client and add tests for all personalizer v1.0 apis . Azure Specs PR - https://github.com/Azure/azure-rest-api-specs/pull/7624

dwaijam opened the pull request:
Generate client and add tests for all personalizer v1.0 apis

dwaijam committed:
generate new apis

dwaijam committed:
Add tests for all personalizer apis

dwaijam committed:
merge master

shahabhijeet commented:
@dwaijam before this can be reviewed, can you add link to your REST api spec.
Also make sure you use generate.ps1 to generate your SDK, that ensures correct meta data generation, it is especially required for properly tagging your generated nuget package.

shahabhijeet commented on src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/ConfigurationsTests.cs:
```diff
@@ -0,0 +1,123 @@
+
+                PolicyContract policy = await client.Policy.ResetAsync();
+
+                Assert.Equal("--cb_explore_adf --quadratic GT --quadratic MR --quadratic GR --quadratic ME --quadratic OT --quadratic OE --quadratic OR --quadratic MS --quadratic GX --ignore A --cb_type ips --epsilon 0.2",
```
nit: make this const

dwaijam replied:
wont fix

shahabhijeet commented on src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/ConfigurationsTests.cs:
```diff
@@ -0,0 +1,123 @@
+                var policy = new PolicyContract
+                {
+                    Name = "app1",
+                    Arguments = "--cb_explore_adf --quadratic GT --quadratic MR --quadratic GR --quadratic ME --quadratic OT --quadratic OE --quadratic OR --quadratic MS --quadratic GX --ignore A --cb_type ips --epsilon 0.2"
```
nit: make this const

dwaijam replied:
wont fix

shahabhijeet commented on src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Personalizer/EvaluationsTests.cs:
```diff
@@ -0,0 +1,89 @@
+            using (MockContext.Start(this.GetType().FullName))
+            {
+                HttpMockServer.Initialize(this.GetType().FullName, "GetEvaluations");
+
```
nit: any reason for adding blank lines? If there isn't a particular reason you should remove it.
When styleCop or static analyzers are enabled, you will be forced to fix this as all these warnings will be reported as errors.

dwaijam replied:
fixed

shahabhijeet commented on src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer.Tests/Properties/launchSettings.json:
```diff
@@ -0,0 +1,11 @@
+      "commandName": "Project",
+      "environmentVariables": {
+        "Key": "Value",
+        "AZURE_TEST_MODE": "Record"
```
@dwaijam can you clarify why are you adding this file?
Are you setting testframework in record mode everytime you run these tests?
We do not want tests to run in record mode during CI
If you want more information, please follow testFramework documentation as these settings are only meant during development of tests and recording of tests locally.

dwaijam replied:
removed

annelo-msft commented:
It looks like the file path for these files is incorrect ... they should live alongside the existing Personalizer files at https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/cognitiveservices/Personalizer

AlexGhiondea commented on src/SDKs/CognitiveServices/dataPlane/Personalizer/Personalizer/Customizations/PersonalizerClientExtensions.cs:
```diff
@@ -0,0 +1,89 @@
+using Microsoft.Azure.CognitiveServices.Personalizer.Models;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
```
I think we should add a copyright header to these files.

dwaijam replied:
fixed

dwaijam committed:
Merge branch 'master' into dwmukh/generatestableclient

dwaijam committed:
resolve all comments

dwaijam committed:
remove csv files

dwaijam committed:
resolve long name errors

AlexGhiondea reviewed (changes requested):
There was some big changes between the time i signed off and the latest version. I have left a couple of questions.

dwaijam commented:
> There was some big changes between the time i signed off and the latest version. I have left a couple of questions.

Addressed your concerns. This is a new version of the API. We are moving from preview to v1.0 and these changes are coming from generating the code from a new swagger which has been approved by the Azure API review board
